protos/camera_server/camera_server.proto

```diff
@@ -0,0 +1,118 @@
+syntax = "proto3";
+
+package mavsdk.rpc.camera_server;
+
+import "mavsdk_options.proto";
+
+option java_package = "io.mavsdk.camera_server";
+option java_outer_classname = "CameraServerProto";
+
+// Provides handling of camera trigger commands.
+service CameraServerService {
+    // Sets the camera information
+    rpc SetInformation(SetInformationRequest) returns(SetInformationResponse) { option (mavsdk.options.async_type) = SYNC; }
+    // Sets the camera capture status
+    rpc SetInProgress(SetInProgressRequest) returns(SetInProgressResponse) { option (mavsdk.options.async_type) = SYNC; }
+
+    // Subscribe to single-image capture commands
+    rpc SubscribeTakePhoto(SubscribeTakePhotoRequest) returns(stream TakePhotoResponse) { option (mavsdk.options.async_type) = ASYNC; }
+    
+    // Adds a photo to the list of available photos
+    rpc PublishPhoto(PublishPhotoRequest) returns(PublishPhotoResponse) { option (mavsdk.options.async_type) = SYNC; }
+}
+
+message SetInformationRequest {
+    Information information = 1; // information about the camera
+}
+
+message SetInformationResponse {
+    CameraServerResult camera_server_result = 1;
+}
+
+message SetInProgressRequest {
+    bool in_progress = 1; // true if capture is in progress or false for idle.
+}
+
+message SetInProgressResponse {
+    CameraServerResult camera_server_result = 1;
+}
+
+message SubscribeTakePhotoRequest {}
+
+message TakePhotoResponse {
+    CameraServerResult camera_server_result = 1;
+    int32 index = 2;
+}
+
+message PublishPhotoRequest {
+    CaptureInfo capture_info = 1;
+}
+
+message PublishPhotoResponse {
+    CameraServerResult camera_server_result = 1;
+}
+
+// Type to represent a camera information.
+message Information {
+    string vendor_name = 1; // Name of the camera vendor
+    string model_name = 2; // Name of the camera model
+    float focal_length_mm = 3; // Focal length
+    float horizontal_sensor_size_mm = 4; // Horizontal sensor size
+    float vertical_sensor_size_mm = 5; // Vertical sensor size
+    uint32 horizontal_resolution_px = 6; // Horizontal image resolution in pixels
+    uint32 vertical_resolution_px = 7; // Vertical image resolution in pixels
+}
```

Comment:
One thing @dayjaby mentioned already in https://github.com/dlech/MAVSDK/pull/1 is that this list is incomplete. I don't remember why I did it this way, but a complete list would look like:

```proto
message Information {
    string vendor_name = 1; // Name of the camera vendor
    string model_name = 2; // Name of the camera model
    uint32 firmware_version = 3; // Version of the camera firmware
    float focal_length_mm = 4; // Focal length
    float sensor_size_h_mm = 5; // Horizontal sensor size
    float sensor_size_v_mm = 6; // Vertical sensor size
    uint32 resolution_h_px = 7; // Horizontal image resolution in pixels
    uint32 resolution_v_px = 8; // Vertical image resolution in pixels
    uint32 lens_id = 9; // Lens ID
    uint32 cam_definition_version = 10; // Camera definition version (iteration)
    string cam_definition_uri = 11; // Camera definition URI (http or mavlink ftp)
}
```

Reply:
Is adding new fields like this in the future considered a breaking change? Or will existing code just set the new fields to the default value?

Reply:
As far as I remember from protobuf, is that it has excellent downwards compatibility as long as you only **add** new fields and don't remove or re-order already existing fields.

Reply:
I think I remember why I left stuff out now. It looks like I only included the fields that are in the Information type of `camera.proto`. This made me assume that the other fields aren't really used in practice also the fact that the mavlink spec called the lens id "reserved").

Since the camera definition file is optional, I'm wondering if it would make sense to have a separate method for specifying that.

Reply:
I guess it makes sense to encourage people to use and provide the camera definition file. When this plugin is ready, it's easier than ever to provide all the functionalities necessary for a camera definition file (e.g. reacting on parameter changes)